src/PIL/GifImagePlugin.py

```diff
@@ -309,6 +313,17 @@ def _seek(self, frame):
         if self.palette:
             self.mode = "P"
 
+    def load_prepare(self):
+        super(GifImageFile, self).load_prepare()
+
+        if self.__frame == 0:
+            # On the first frame, clear the buffer.  Use the transparency index
+            # if we have one, otherwise use the background index.
+            default_color = self.info.get(
+                "transparency", self.info.get("background", 0)
+            )
+            self.im.paste(default_color, (0, 0, self.size[0], self.size[1]))
```

Comment:
```suggestion
        if not self.im:
            # On the first frame, clear the buffer.  Use the transparency index
            # if we have one, otherwise use the background index.
            default_color = self.info.get(
                "transparency", self.info.get("background", 0)
            )
            self.im = Image.core.fill(self.mode, self.size, default_color)

        super(GifImageFile, self).load_prepare()
```
This saves the parent `load_prepare` from creating an image only to have it overwritten here. Instead, this just creates the image and the parent does not.

Reply:
I've created #5557 with this functionality, minus the background color fallback.